### PR TITLE
Fluid typography with attention to vertical rhythm

### PR DESCRIPTION
**Pull request by vwong**
Igor,

I've incorporated your feedback and added more refinements, as well as fixed a few more bug.  I've added a whole new [hidden] page at http://localhost:3000/typography.html, which showcases the implementation details and discusses some design ideas.  Let me know what you think.

Voon

### CODE REVIEW

**Comment by igoradamenko** [comment]
Hey Voon!

It looks much better! I like it! And your new page is nice, I think it shouldn't be hidden.

But I still have some comments 🐨

1. `Yeah! They have absolutely no custom styles :)` ← I think we should remove or rewrite this phrase about `blockquote`. Now it has some margin / padding, and we should be honest.

2. I'm not sure that `dl` should have margin-left. Is this side effect from `ol` / `ul` margins? Maybe it must be 0?

3. Is it possible to vertically align `hr`? Now it is not: 
screen shot 2016-11-10 at 02 32 15

    Also can we change it's color to lighter? Sections separated by space between them and `hr` is just a note: “This space is not a mistake, it's by design”. So when color of `hr` is so dark, note looks like: “THIS SPACE IS NOT A MISTAKE! REALLY! IT'S BY DESIGN!”. I think we shouldn't shout at the user ^_^

4. If forms ruin rhythm can we leave some controls without changes? Before they were tiny and friendly, like native, but now they are bigger. I'm talking about select, submit & reset buttons. If we remove `height: 2.1rem` from them, they will be better, won't they?

5. In “Misc.” section we have spoiler with next phrase: “This simple spoiler does not work in Internet Explorer and Firefox yet.”, but right now it works in Firefox. And when we set padding, it looks not so good in Firefox. Maybe we should remove all paddings, backgrounds, etc, and create something like this:
screen shot 2016-11-10 at 03 38 48

    `border-bottom` sets only for `details[open]`.

6. When page does not start from section, there isn't enough space between header and main content:
screen shot 2016-11-10 at 03 42 34

7. You are talking about horror of resizing textarea. Maybe `resize: vertical` should be set for textareas?

File: dev/example/typography.html [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      <h1>Just h1 over multiple lines</h1>
+      <p>A short burst of large text in headers tolerate smaller <em>leading</em> than the main text.  Descenders in one line can come close to ascenders on the next without discomfort.  In most cases, and in large enough viewports, <code>h1</code> headers are single lined anyway.</p>
+      <h2>What about h2?</h2>
+      <p>Same thing.</p>
+      <h2>And h2 over multiple lines?  How does it look?</h2>
+      <p>No loss of vertical rhythm.</p>
+      <h3>This is an h3</h3>
+      <p>Still going...</p>
+      <h3>This is an h3 over multiple lines; I struggle to find fluff words to pad it out</h3>
+      <p>It's time for a compromise.  For the <a href="#scale">scale</a> chosen, it is difficult to find a <em>leading</em> that simultaneously sits every line on the baseline (to maintain overall rhythm), and that is not too large or too small.  So, a half-baseline offset is introduced whenever we have an even number of <code>h3</code> lines.  As with the other headers, in most cases, <code>h3</code> headers are single lined anyway.</p>
```
Comment (igoradamenko) [inline dev/example/typography.html]: Reference should be `#scales` 🐨

**Comment by vwong** [comment]
Igor,

Thanks for the feedback, am glad you're liking the changes.  I'll fix it up taking your ideas into consideration - I think I have an idea how.  Will get back to you soon (am only doing this on my commute to/from work).

Voon

File: dev/scss/style.scss [hunk 2]
```diff
@@ -223,6 +224,9 @@ input {
 textarea {
 	@include inputs;
 	height: $baseline*3.5;
```
Comment (igoradamenko) [inline dev/scss/style.scss]: Let's add spaces around math operations? `$baseline * 3.5` is more readable than `$baseline*3.5`.

Comment (igoradamenko) [inline dev/scss/style.scss]: Or you can leave it now. I can fix it later by myself 🐨

**Comment by vwong** [comment]
Igor,

"They have absolutely no custom styles"; I understood this as there is no class/id for styles, since we're styling HTML tags directly.  Up to you, if you want to reword it.

Form buttons & selects: I think should be the same size as all other form elements, such as text inputs.  So, the height should stay.  Most other CSS libs/frameworks do the same.

Using space around math: I'm undecided about this one.  It may look slightly better, but it could end up being ugly or attributes that take multiple space separated values.  For example:

    {
      margin: $variable*2 0;
    }

vs:

    {
      margin: ($variable * 2) 0;
    }

I think the former looks cleaner, but not fussed about it.

All other comments, I think I've already addressed in the code.  I think I'm "done" for now.  Unless you have any further requests?

Voon

**Comment by igoradamenko** [comment]
Thank you, Voon!

**Pull request merged**